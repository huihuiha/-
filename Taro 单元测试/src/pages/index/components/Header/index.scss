.header-wrap {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 100vw;
    height: 100px;
    background-color: beige;

    .input {
        width: 500px;
        padding: 8px;
        border: 1px solid #000;
        border-radius: 5px;
    }
}