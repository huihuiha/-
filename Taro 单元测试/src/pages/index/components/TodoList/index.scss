.todo-list-wrap {
    margin-top: 20px;
    margin: 0 auto;
    width: 100vw;
    background-color: azure;
    .item-wrap {
        display: flex;
        justify-content: center;
        align-items: center;
        border-left: burlywood;
        border-left-width: 15px;
        padding-left: 10px;
        border-left-style: solid;
        .btn {
            display: flex;
            justify-content: center;
            align-items: center;
            height: 50px;
            width: 150px;
            font-size: 30px;
            color: #444;
        }
    }
}