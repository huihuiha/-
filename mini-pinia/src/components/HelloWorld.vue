<script setup>
import { useCounterStore } from '../store/counter';
const counter = useCounterStore();
</script>

<template>
  <h1>{{ counter.count }} --- {{ counter.doubleCount }}</h1>
  <button @click="counter.count++">++</button>
  <button @click="counter.inc()">++</button>
  <!-- 更新多个属性 -->
  <button @click="counter.$patch({ count: counter.count + 1 })">++</button>
  <button @click="counter.$patch((state) => state.count++)">++</button>
</template>

<style scoped>
.read-the-docs {
  color: #888;
}
</style>
