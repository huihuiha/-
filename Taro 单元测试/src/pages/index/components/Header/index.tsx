import { View, Input, Button } from '@tarojs/components';
import { useState } from 'react';
import './index.scss';

const Header = ({
    addItem,
}) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (value) => {
        setInputValue(value);
    }

    const handleAddItem = () => {
        handleInputChange('');
        addItem({ show: true, value: inputValue });
    }

    return (
        <View className='header-wrap'>
            <Input
                className='input'
                type="text"
                placeholder='请输入todo事件'
                onInput={(e) => handleInputChange(e.detail.value)}
                value={inputValue}
            />
            <Button className='btn add-btn' onClick={handleAddItem}>添加</Button>
        </View>
    )
}
export default Header;