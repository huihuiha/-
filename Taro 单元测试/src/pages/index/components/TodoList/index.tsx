import { View, Button } from '@tarojs/components';
import { useMemo } from 'react';
import './index.scss';

const TodoList = ({
    list,
    handleDeleteItem
}) => {
    // 数量展示
    const count = useMemo(() => {
        let count = 0;
        list.forEach(item => {
            if (item.show) {
                count++
            }
        })
        return count;
    }, [list])


    return (
        <View className='todo-list-wrap'>
            {
                list.map((item, index) => (
                    item.show ? (
                        <View key={item} className='item-wrap'>
                            <View className='item' key={item.value}>{item.value}</View>
                            <Button className='btn' onClick={() => handleDeleteItem(index)}>删除</Button>
                        </View>
                    ) : ''
                ))
            }
            <View className="count">{count}</View>
        </View>
    )
}
export default TodoList;