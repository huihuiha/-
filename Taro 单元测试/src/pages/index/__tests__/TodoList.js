// import { render, screen, fireEvent } from '@testing-library/react'
import {shallow} from 'enzyme';
import TodoList from '../components/TodoList/index';
import { Input } from '@tarojs/components';
import toJson from 'enzyme-to-json'
describe('TodoList 列表组件', () => {
  test('TodoList 列表组件样式发生变化的时候，做出提示', () => {
    const wrapper = shallow(<TodoList list={[]} />);
    expect(toJson(wrapper)).toMatchSnapshot()
  });

  test('当 list 属性元素多少项的时候，根据子元素 show属性判断展示多少项', () => {
    const list = [
        {show: true, value: 1},
        {show: false, value: 2},
        {show: true, value: 3},
        {show: true, value: 4}
    ]
    const wrapper = shallow(<TodoList list={list} />);
    const items = wrapper.find('.item');
    expect(items.length).toBe(3)
  });

  test('当有list列表有可以展示的元素 > 0，则下方列表展显示的数量', () => {
    let list = [
        {show: true, value: 1},
        {show: false, value: 2},
        {show: true, value: 3},
        {show: true, value: 4}
    ]
    const wrapper = shallow(<TodoList list={list}/>);
    const countWrapper = wrapper.find('.count');
    expect(countWrapper.props().children).toBe(3);
  });

  test('当list列表没有可展示的元素的时候，下方展示数量为0，且列表没有元素', () => {
    const wrapper = shallow(<TodoList list={[]}/>);
    expect(+wrapper.find('.count').text()).toBe(0);
    expect(wrapper.find('.item').length).toBe(0);
  });

  test('接收父组件传递过来的删除事件 handleDeleteItem，点击元素后面的删除按钮，则执行', () => {
    let list = [
        {show: true, value: 1},
        {show: true, value: 2},
        {show: true, value: 3},
        {show: true, value: 4}
    ]
    const fn = jest.fn()
    const wrapper = shallow(<TodoList list={list} handleDeleteItem={fn} />);
    const btns = wrapper.find('.btn');
    btns.at(1).simulate('click');
    expect(fn).toBeCalledTimes(1)
  });
})