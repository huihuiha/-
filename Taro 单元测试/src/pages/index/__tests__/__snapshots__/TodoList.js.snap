// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList 列表组件 TodoList 列表组件样式发生变化的时候，做出提示 1`] = `
<ForwardRef
  className="todo-list-wrap"
>
  <ForwardRef
    className="count"
  >
    0
  </ForwardRef>
</ForwardRef>
`;
